config/overrides.go

```diff
@@ -361,3 +363,84 @@ func NamePrefixRemoval() tjconfig.ResourceOption {
 		r.ExternalName.OmittedFields = append(r.ExternalName.OmittedFields, "name_prefix")
 	}
 }
+
+// KnownReferencers adds referencers for fields that are known and common among
+// more than a few resources.
+func KnownReferencers() tjconfig.ResourceOption { //nolint:gocyclo
+	return func(r *tjconfig.Resource) {
+		for k, s := range r.TerraformResource.Schema {
+			if s.Computed && !s.Optional {
+				continue
+			}
+			switch {
+			case strings.HasSuffix(k, "role_arn"):
+				r.References[k] = tjconfig.Reference{
+					Type:      "github.com/crossplane-contrib/provider-jet-aws/apis/iam/v1alpha2.Role",
+					Extractor: common.PathARNExtractor,
+				}
+			case strings.HasSuffix(k, "security_group_ids"):
+				r.References[k] = tjconfig.Reference{
+					Type:              "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.SecurityGroup",
+					RefFieldName:      strings.TrimSuffix(name.NewFromSnake(k).Camel, "s") + "Refs",
+					SelectorFieldName: strings.TrimSuffix(name.NewFromSnake(k).Camel, "s") + "Selector",
+				}
+			}
+			switch k {
+			case "vpc_id":
+				r.References["vpc_id"] = tjconfig.Reference{
+					Type: "github.com/crossplane-contrib/provider-jet-aws/apis/ec2/v1alpha2.VPC",
```

Comment:
Can we handle resolver generation if the referrer and the referee are in the same package?

Reply:
`angryjet` takes care of the case where the type is in the same package even if a full path is given. Though I'm not really sure what's going on here in these two. It prints an error talking about import cycle with autoscaling and for some reason adds an import to self. But I didn't have the patience to dig too deep.